pkg/database/fake/database_generated.go

```diff
@@ -152,11 +161,18 @@ func (f *Database) filterResourceByKindApiVersionNamespaceAndName(kind, apiVersi
 }
 
 func (f *Database) WriteResource(_ context.Context, k8sObj *unstructured.Unstructured, _ []byte) error {
+	if f.err != nil {
+		return f.err
+	}
 	f.resources = append(f.resources, k8sObj)
 	return nil
 }
 
 func (f *Database) WriteUrls(_ context.Context, k8sObj *unstructured.Unstructured, jsonPath string, logs ...models.LogTuple) error {
+	if f.urlErr != nil {
+		return f.urlErr
+	}
```

Comment:
Why even make this call if there is a known error with the database?

Reply:
This condition is used in the unit tests for the sink to simulate `WriteUrls` returning an error so that we can test that the sink responds with the correct http status codes

Reply:
yeah, `pkg/database/fake` is a database mock for testing